src/plugins/MamicoCoupling.cpp

```diff
@@ -0,0 +1,80 @@
+#include "MamicoCoupling.h"
+#include "Domain.h"
+
+#ifdef MAMICO_COUPLING
+#include <coupling/interface/impl/ls1/LS1MamicoCouplingSwitch.h>
+//#include <LS1MamicoCouplingSwitch.h>
+#endif
+
+void MamicoCoupling::readXML(XMLfileUnits& xmlconfig)
+{	
+    return;
+}
+
+void MamicoCoupling::init(ParticleContainer* particleContainer,
+		DomainDecompBase* domainDecomp, Domain* domain)
+{
+    #ifdef MAMICO_COUPLING
+    if(_macroscopicCellService==NULL) 
```

Comment:
Either use the keyword `nullptr` instead of the macro `NULL` or directly use the value of the pointer for the if:
```suggestion
    if(not _macroscopicCellService) 
```